addons/stock/tests/test_product.py

```diff
@@ -81,3 +81,63 @@ def test_free_quantity(self):
         self.picking_out_2.do_unreserve()
         # 8 units are available again
         self.assertAlmostEqual(40.0, self.product_3.free_qty)
+
+    def test_search_product_template(self):
+        """
```

Comment:
Why is this test in the stock module? You seem to c/p existing test code from product, and you don't seem to use anything defined in the stock module :thinking:

Reply:
Because I wanted a use case where a variant is archived (not deleted) by Odoo and not by the user

See the test description:
> Suppose a variant V01 that can not be deleted because it is used by a
lot [1]. Then, the variant's template T is changed: we add a dynamic
attribute. Because of [1], V01 is archived.

So, when adding an attribute, it leads to:
https://github.com/odoo/odoo/blob/5ff4eb022757d7a522ccb975f8eb64053ef9780b/addons/product/models/product_template.py#L636-L637
where some variants X will be deleted and some other Y will be archived. The filter `_filter_to_unlink` is used to find X and Y. For instance, in case the variant is linked to a lot, this variant will be included in Y:
https://github.com/odoo/odoo/blob/d6bd78e578104f43318fbe3c11f325063e3e5042/addons/stock/models/product.py#L553-L557

That's the reason why I wrote a test here